### ISSUE TITLE
Search on bi_fm_index doesn't find all matches

### ISSUE THREAD

**Issue by SGSSGene** [seqan/seqan3#1315]
## Platform

  - SeqAn version: seqan3 - commit f436e1c53d89451791bb6bfc0029becbff7c3fbd (master) - happens with older version too.
  - Operating system: Linux schenker 5.3.6-arch1-1-ARCH #1 SMP PREEMPT Fri Oct 11 18:28:05 UTC 2019 x86_64 GNU/Linux
  - Compiler: gcc 9.2.0

## Description
Using a bi_fm_index with optimum search schemes, doesn't return all matches.

### How to repeat the problem
Lets assume we have a bi_fm_index over the text `"ACGATACG"_dna4` and we are searching for `"ACGACACG"_dna4`. Notice that the 5. character is changed.
If we now search with max 1 error we would expect one result.

The following complete examples shows the issue in action:
```
#include <seqan3/core/debug_stream.hpp>
#include <seqan3/search/algorithm/search.hpp>
#include <seqan3/search/fm_index/bi_fm_index.hpp>

using namespace seqan3;
int main(int argc, char * argv[])
{
    auto genome  = std::vector<std::vector<seqan3::dna4>> {"ACGATACG"_dna4};
    auto index   = seqan3::bi_fm_index{genome};
    auto pattern = "ACGACACG"_dna4;

    configuration const cfg     = search_cfg::max_error{search_cfg::total{1},
                                                    search_cfg::substitution{1},
                                                    search_cfg::insertion{0},
                                                    search_cfg::deletion{0}};
    auto hits = search(pattern, index, cfg);

    debug_stream << "result count: " << hits.size() << "\n";

    for (auto [seq, pos] : hits) {
        debug_stream << "seq: " << seq << " at pos " << pos << "\n";
    }

    // expected output:
    //  result count: 1
    //  seq 0 at pos 0

    return 0;
}
```

### Expected behaviour
I expected to find the pattern at position 0.

### Actual behaviour
I don't find the pattern.

**Comment by eseiler** [comment]
Thank you for reporting!

This should be fixed by #1316 

When dealing with text collections, we currently just merge the texts with a delimiter. We implicitly expect to encounter at least 1 delimiter in the modified text.
If we only have one text in the collection, we need to append 1 delimiter at the end, since merging 1 text with a delimiter just returns the text without delimiter.

**Comment by SGSSGene** [comment]
Wow! Thank you. This was a fast fixed. I will test it tomorrow. Does this fix means I have to rebuild my indices?

**Comment by eseiler** [comment]
> Does this fix means I have to rebuild my indices?

Yes, unfortunately. But only for those indices where you built the index over a text collection, but only had one element in the text collection, i.e. like your example.
If you had multiple entries in the collection, e.g. `auto genome  = std::vector<std::vector<seqan3::dna4>> {{"ACGTACGTC"_dna4}, {"ACGATACG"_dna4}};`, the index remains the same and you do not have to rebuild.
